transcom/mymove
# B-21492 Admin User can assign Secondary Transportation Office Assignments

## [B-21492](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21492)

## Summary

Allows admin users to assign up to two transportation offices to office users.

- Adds TransportationOfficeAssignments slice to OfficeUser model
- Adds TransportationOfficeAssignments slice to UserIdentify model
- Adds TransportationOfficeAssignments array to admin create/update office user payloads
- Remove TransportationOfficeId from admin CREATE/UPDATE (but not GET) office user payloads
- Adds TransportationOfficeAssignments array to admin GET office user payloads
- ...

Hopefully this sets up the follow on task to be pretty easy.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the admin app.
2. Login as a new or existing admin user.
3. Create a new office user, expect the UI for assigning a transportation office to be slightly different.
4. Test out any combination of one or two transportation office assignments then modify by adding/remove/changing assignments a few times.
5. Ensure the designated primary office is displayed correctly on the list of office user and that the assignments you created display as you created them when you re-open the office user form.
6. Try to break it. The validation message refresh is a bit wonky but that seems to be a quirk of the react-admin library. That shouldn't affect your ability to submit valid data or not submit invalid data.

If you wish to verify the creation of your transportation office assignments in the DB the following query may be helpful.
`SELECT * FROM public.transportation_office_assignments toas
JOIN transportation_offices tos ON toas.transportation_office_id = tos.id
JOIN office_users ou ON ou.id = toas.id
WHERE ou.last_name = 'O''Testoghue'`

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="559" alt="Screenshot 2024-10-29 at 6 03 40 PM" src="https://github.com/user-attachments/assets/15c76f90-0979-4dea-8e26-29cdf9786af3">
